src/commands/startSsh.ts

```diff
@@ -0,0 +1,114 @@
+        await startSshInternal(node);
+    } catch (error) {
+        sshSessionsMap.set(node.root.client.fullName, { starting: false, terminal: undefined, tunnel: undefined, localPort: undefined });
+        throw error;
+    }
+}
+
+async function startSshInternal(node: SiteTreeItem): Promise<void> {
+    const siteClient: SiteClient = node.root.client;
+    if (!siteClient.isLinux) {
+        throw new Error('Azure SSH is only supported for Linux web apps.');
+    }
+
+    await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification }, async (progress: vscode.Progress<{}>): Promise<void> => {
+
+        remoteDebug.reportMessage('Checking app settings...', progress);
+
+        const confirmDisableMessage: string = 'Remote debugging must be disabled in order to SSH. This will restart the app.';
+        const siteConfig: SiteConfigResource = await siteClient.getSiteConfig();
+        // remote debugging has to be disabled in order to tunnel to the 2222 port
+        await remoteDebug.setRemoteDebug(false, confirmDisableMessage, undefined, siteClient, siteConfig, progress);
+
+        remoteDebug.reportMessage('Initializing SSH...', progress);
+        const publishCredential: User = await siteClient.getWebAppPublishCredential();
+        const localHostPortNumber: number = await portfinder.getPortPromise();
+        // should always be an unbound port
+        const sshPortNumber: number = 2222;
+        const tunnelProxy: TunnelProxy = new TunnelProxy(localHostPortNumber, sshPortNumber, siteClient, publishCredential);
+
+        await tunnelProxy.startProxy();
+
+        remoteDebug.reportMessage('Connecting to SSH...', progress);
+        await connectToTunnelProxy(node, tunnelProxy, localHostPortNumber);
+    });
+}
+
+async function connectToTunnelProxy(node: SiteTreeItem, tunnelProxy: TunnelProxy, port: number): Promise<void> {
+    const sshTerminalName: string = `${node.root.client.fullName} - SSH`;
+    // -o StrictHostKeyChecking=no doesn't prompt for adding to hosts
+    // -o "UserKnownHostsFile /dev/null" doesn't add host to known_user file
+    // -o "LogLevel ERROR" doesn't display Warning: Permanently added 'hostname,ip' (RSA) to the list of known hosts.
+    const sshCommand: string = `ssh -c aes256-cbc -o StrictHostKeyChecking=no -o "UserKnownHostsFile /dev/null" -o "LogLevel ERROR" root@127.0.0.1 -p ${port}`;
+
+    // if this terminal already exists, just reuse it otherwise create a new terminal.
+    // tslint:disable-next-line:strict-boolean-expressions
+    const terminal: vscode.Terminal = vscode.window.terminals.find((activeTerminal: vscode.Terminal) => { return activeTerminal.name === sshTerminalName; }) || vscode.window.createTerminal(sshTerminalName);
+
+    // because the container needs time to respond, there needs to be a delay between connecting and entering password
+    terminal.sendText(sshCommand, true);
+    await delay(3000);
```

Comment:
So this is a workaround, right? That'll be fixed when we switch to tasks? If so - please file an issue and add a comment in the code linking to that issue